JSIL/ILBlockTranslator.cs

```diff
@@ -1749,6 +1749,23 @@ bool forDynamic
                     }
                 }
             }
+            else if (
```

Comment:
Also, I seem to remember ILBlockTranslator already having some fairly general logic for dealing with goofy comparisons/arithmetic on nulls, so if this could sit next to that existing stuff or replace it, that would be awesome.